File: app/src/main/java/com/sorcerer/sorcery/iconpack/ui/theme/BaseTheme.java
package com.sorcerer.sorcery.iconpack.ui.theme;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2017/3/12
 */

public abstract class BaseTheme {

    public abstract boolean isDark();

    abstract String name();

    abstract int style();

    abstract int primaryColor();

    abstract int primaryDarkColor();

    abstract int accentColor();

    abstract int backgroundColor();

    abstract int cardColor();

    abstract int primaryTextColor();

    abstract int secondaryTextColor();
}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/ui/theme/BaseTheme.kt
package com.sorcerer.sorcery.iconpack.ui.theme

/**
 * @description:
 * *
 * @author: Sorcerer
 * *
 * @date: 2017/3/12
 */

abstract class BaseTheme {

    abstract val isDark: Boolean

    abstract fun name(): String

    abstract fun style(): Int

    abstract fun primaryColor(): Int

    abstract fun primaryDarkColor(): Int

    abstract fun accentColor(): Int

    abstract fun backgroundColor(): Int

    abstract fun cardColor(): Int

    abstract fun primaryTextColor(): Int

    abstract fun secondaryTextColor(): Int
}
